enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,476 @@
+
+1. We need to update the microshift-ovs-init.service to 
+   * Create br-ex bridge, and assign a static IP to it.
+   * Since we don't need NetworkManager to manage the DHCP client. We can remove
+   those logic from the configure-ovs.sh.
+   * For upgrading, we also need to remove the NetworkManager connections
+   created by configure-ovs.sh in previous version.
+
+### Risks and Mitigations
+
+The OVN-Kubernetes configuration described in this document is not used by any
+other known platforms. Therefore there is no CI with this setup in
+OVN-Kubernetes upstream. In order to minimize the risk of regression, we need to
+improve the OVN-Kubernetes CI to cover this setup.
+
+### Drawbacks
+
+The egress features that rely on sending packets to wire without passing kernel
+will not work. As there is no uplink port to forward those packets to. So
+egressGW and egressIP will not work. A possible solution is to forward those
+traffic to the host kernel, then adding iptables rules to steer the traffic to
+wire.
+
+With this change, the ovn node annotations such
+`k8s.ovn.org/node-primary-ifaddr` and `k8s.ovn.org/l3-gateway-config` will use
+the local IP of br-ex as the node IP, instead of the IP of host's primary
+interface. Therefore the following features that need to read these node
+annotation will be affected. The egress rules of Egress firewall contains the
+node IP. And the traffic case of services with `internalTrafficPolicy=Local`
+backed by hostNetwork pods will break. For services with
+`internalTrafficPolicy=Local`, ovn-k will only put local host endpoints to the
+node switch loadbalancer. If the pod is hostNetwork, the local endpoint means
+the pod with the NodeIP. Normally, the ovn-k will use the nodeIP as the ovn-k
+external IP in node annotation `k8s.ovn.org/l3-gateway-config`. However, in our
+case, as we use a dummy interface for br-ex, we have different IPs for nodeIP
+and ovn-k external IP. So ovn-k cannot identify the right local endpoint in such
+a case. It then put no endpoint to the load-balancer.This problem is not so
+critical to MicroShift. Since we only get one node in the cluster. Creating a
+service with `internalTrafficPolicy=Local` makes no sense in such a case. We can
+fix this problem later until we find a proper solution.
+
+However, the aforementioned features/traffic-case are not required in
+MicroShift. So we want to make the work first and leave advanced features till
+later.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. We'll need to allocate a IP address to br-ex bridge. This IP address will not
+   be exposed externally. Shall we use a fixed IP address, or we allow user to
+   be able to specify it?
+
+   A: We decide to reuse the `advertise-address` of kubernetes API server by add
+   this address to br-ex. Users can specify the address with the config flag
+   `apiServer.advertiseAddress` in microshift. The default value is `10.44.0.0`.
```

Comment:
but this isnt a valid ip address.

Reply:
ok so its legal, but most people dont use it because its most commonly used as the network address for a subnet (along with .255 as the broadcast)